### DISCUSSION TITLE
Dark mode

### DISCUSSION THREAD

**Discussion by Aster89** [qutebrowser/qutebrowser#5733] [Q&A] [answered]
I've seen several opened/closed issues regarding, at least marginally, dark mode, but I still have to understand if it is something that can be set by me, or that only works for sites that have provided CSS sheets that define a darkmode. I've also seen that there are several options like `colors.webpage.darkmode.*`, but changing them seems to have no effect.

**Comment by The-Compiler** [comment, marked as answer]
There are generally four ways to get dark websites:

- `colors.webpage.prefers_color_scheme_dark` which asks the page for a dark stylesheet, so the page will need to provide a dark CSS (requires QtWebEngine, Qt 5.14 and a restart after setting)
- `colors.webpage.darkmode.*` which adds a "filter" to make all websites dark. It also requires QtWebEngine, Qt 5.14 and a restart. If changing it didn't have any effect, I'm guessing you didn't restart qutebrowser or you have an older/different backend.
- `content.user_stylesheets` which can be used with e.g. [solarized everything CSS](https://github.com/alphapapa/solarized-everything-css/) to restyle websites - it generally doesn't work as well as the above, but it works with all Qt versions and can be set at runtime
- Use (a stripped down version of) darkreader via qutebrowser's Greasemonkey support, see https://github.com/darkreader/darkreader/issues/926#issuecomment-575893299 - I've never tested this myself though.

**Reply by Aster89** [reply]
This is what is in my `autoconfig.yml`:
```none
# If a config.py file exists, this file is ignored unless it's explicitly loaded
# via config.load_autoconfig(). For more information, see:
# https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/configuring.asciidoc#loading-autoconfigyml
# DO NOT edit this file by hand, qutebrowser will overwrite it.
# Instead, create a config.py - see :help for details.

config_version: 2
settings:
  colors.webpage.darkmode.enabled:
    global: true
  colors.webpage.darkmode.policy.page:
    global: always
  colors.webpage.prefers_color_scheme_dark:
    global: true
```
which means some of the settings are in effect. However, the very page where I'm writing is still the same mostly white page as before.

As regards my version of Qt, it seems to be recent enough from [`:version`](https://paste.the-compiler.org/view/f6495cf2).

What do you think?

**Reply by The-Compiler** [reply]
Your version info says:

> Autoconfig loaded: no
> Config.py: /home/enrico/.config/qutebrowser/config.py has been loaded

Please read the first couple of lines of the file you quoted.

**Reply by Aster89** [reply]
Oops. Thank you very much :P

**Comment by Aster89** [comment]
My conclusion is that the 3rd solution, settting `content.user_stylesheets` to a nice CSS stylesheet, is the best solution, exactly for the reason that it can be changed at run time, which is so important when a website is so crappy that with a dark CSS it becomes unreadable.

**Comment by goyalyashpal** [comment]
> * `colors.webpage.prefers_color_scheme_dark` which asks the page for a dark stylesheet, so the page will need to provide a dark CSS (requires QtWebEngine, Qt 5.14 and a restart after setting)

it ain't working. 
the documentation website (local) made using sphinx with shibuya theme (which does have 3 modes: system, force light, force dark) - stays in light mode despite changing the system theme and qute restart.

the system mode of website is working well in normal browsers.

i have removed the config.py before doing all these restarts etc, and here's the autoconfig.yaml:

```console
$ ls "$APPDATA"/qutebrowser/config/
autoconfig.yml  bookmarks  greasemonkey  quickmarks

$ bat -pP "$APPDATA"/qutebrowser/config/autoconfig.yml
# If a config.py file exists, this file is ignored unless it's explicitly loaded
# via config.load_autoconfig(). For more information, see:
# https://github.com/qutebrowser/qutebrowser/blob/main/doc/help/configuring.asciidoc#loading-autoconfigyml
# DO NOT edit this file by hand, qutebrowser will overwrite it.
# Instead, create a config.py - see :help for details.

config_version: 2
settings:
  auto_save.session:
    global: true
  qt.highdpi:
    global: true
```

```
qutebrowser v3.1.0
Backend: QtWebEngine 6.6.1, based on Chromium 112.0.5615.213 (from api)
Qt: 6.6.1

CPython: 3.12.0
PyQt: 6.6.1

```


also, there's no `qutebrowser.exe --version`  on cli?

**Reply by The-Compiler** [reply]
Please open a new issue and also try with `--temp-basedir` - what you're describing here is unrelated to this discussion, and even seems to be unrelated to the setting you're quoting (given that you're not actually setting it). Note that the setting also was renamed in the meantime.

**Reply by goyalyashpal** [reply]
> unrelated to the setting you're quoting (given that you're not actually setting it).

i am not setting the setting as it's (supposed to be) at auto by default, which means follow system which is what i need
https://qutebrowser.org/doc/help/settings.html#colors.webpage.preferred_color_scheme


> Please open a new issue and also try with `--temp-basedir`

ohkay, i will
